### PR TITLE
Tracking Bitcoin Price

### PR DESCRIPTION
**Pull request by Azelphur**
Hi,

I wrote this primarily for myself, but I figured I'd clean it up a bit, perhaps others could make use of it. It allows you to put a note on the account in the format BTC:x (eg BTC:0.01) and it will create transactions to keep that accounts balance correct to the amount of BTC you have.

### CODE REVIEW

File: sync-bitcoin.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    return undefined;
+  }
+}
+
+(async () => {
+  await openBudget();
+  const bitcoinPrice = await getBitcoinPrice();
+  if (!bitcoinPrice) {
+    throw new Error("Unable to retrieve Bitcoin price. Check your BITCOIN_PRICE_URL and BITCOIN_PRICE_JSON_PATH environment variables");
+  }
```
Comment (psybers) [inline sync-bitcoin.js:45]: Do this after getting the bc price (so if that fails, the budget file was not opened).

```suggestion
  const bitcoinPrice = await getBitcoinPrice();
  if (!bitcoinPrice) {
    throw new Error("Unable to retrieve Bitcoin price. Check your BITCOIN_PRICE_URL and BITCOIN_PRICE_JSON_PATH environment variables");
  }
  await openBudget();
```

File: sync-bitcoin.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        return acc;
+    }, obj);
+}
+
+async function getBitcoinPrice() {
+  let response = undefined;
+  const url = process.env.BITCOIN_PRICE_URL || "https://api.kraken.com/0/public/Ticker?pair=xbtusd"
+  const path = process.env.BITCOIN_PRICE_JSON_PATH || "result.XXBTZUSD.c[0]"
+  try {
+    response = await fetch(url);
```
Comment (psybers) [inline sync-bitcoin.js:32]: Response can be defined below.

```suggestion
  const url = process.env.BITCOIN_PRICE_URL || "https://api.kraken.com/0/public/Ticker?pair=xbtusd"
  const path = process.env.BITCOIN_PRICE_JSON_PATH || "result.XXBTZUSD.c[0]"
  try {
    const response = await fetch(url);
```

File: README.md [hunk 3]
```diff
@@ -252,3 +257,17 @@ node track-investments.js
 
 It is recommended to run this script once per month.
 
+### Tracking the Bitcoin Price
+This script tracks the value of Bitcoin. It adds new transactions to keep the account balance equal to the latest value.
+
+There is one tag you can set in the account notes, BTC:X, where X is the number of Bitcoin you own, eg `BTC:0.01`
+
+You can optionally change the API endpoint used to retrieve the Bitcoin price, an example for retrieving the price in GBP is `BITCOIN_PRICE_URL="https://api.kraken.com/0/public/Ticker?pair=xbtgbp"` `BITCOIN_PRICE_JSON_PATH=".result.XXBTZGBP.c[0]"`
```
Comment (psybers) [inline README.md:265]: Please wrap at column 80.

File: README.md [hunk 4]
```diff
@@ -252,3 +257,17 @@ node track-investments.js
 
 It is recommended to run this script once per month.
 
+### Tracking the Bitcoin Price
```
Comment (psybers) [inline README.md:260]: Be sure to add a link in the TOC above (around line 13).

**Review by psybers — CHANGES_REQUESTED** [review]
Great, thank you for this!

**Comment by Azelphur** [comment]
Thanks for the feedback, all good suggestions :) I believe I've corrected them all now.

**Review by psybers — APPROVED** [review]

**Pull request merged**